### PR TITLE
v0.3.1: Göttingen release (aka June release)

### PR DESCRIPTION
**Pull request by LucasSte**
Release tests are running [here](https://github.com/LucasSte/solang/actions/runs/5395030402).

### CODE REVIEW

**Review by seanyoung — CHANGES_REQUESTED** [review]

**Review by xermicus — APPROVED** [review]

**Review by seanyoung — APPROVED** [review]

**Comment by LucasSte** [comment]
I'm waiting for the release test build to finish before merging: https://github.com/LucasSte/solang/actions/runs/5424128921

**Pull request merged**
